eva-pipeline/src/test/java/embl/ebi/variation/eva/pipeline/steps/VariantsAnnotGenerateInputBatchTest.java

```diff
@@ -0,0 +1,294 @@
+            reader.close();
+        }
+
+        outputFile.delete();
+    }
+
+    @Test
+    public void variantReaderShouldReadVariantsWithoutAnnotationField() throws Exception {
+        insertDocuments();
+        MongoItemReader<DBObject> mongoItemReader = variantsAnnotGenerateInputBatch.initReader(collectionName, new MongoTemplate(mongoClient, databaseName));
+        mongoItemReader.open(executionContext);
+
+        int itemCount = 0;
+        DBObject doc;
+        while((doc = mongoItemReader.read()) != null) {
+            itemCount++;
+            assertTrue(doc.containsField("chr"));
+            assertTrue(doc.containsField("start"));
+            assertFalse(doc.containsField("annot"));
+        }
+        assertEquals(itemCount, 1);
+        mongoItemReader.close();
+    }
+
+    @Test
+    public void vepInputLineProcessorShouldConvertAllFieldsInVariant() throws Exception {
+        DBObject dbo = constructDbo(variantWithoutAnnotation);
+
+        ItemProcessor<DBObject, VariantWrapper> processor =  new VariantAnnotationItemProcessor();
+        VariantWrapper variant = processor.process(dbo);
+        assertEquals("+", variant.getStrand());
+        assertEquals("20", variant.getChr());
+        assertEquals("G/A", variant.getRefAlt());
+        assertEquals(60343, variant.getEnd());
+        assertEquals(60343, variant.getStart());
+    }
+
+    @Test
+    public void vepInputWriterShouldWriteAllFieldsToFile() throws Exception {
+        DBObjectToVariantConverter converter = new DBObjectToVariantConverter();
+        VariantWrapper variant = new VariantWrapper(converter.convertToDataModelType(constructDbo(variantWithAnnotation)));
+
+        FlatFileItemWriter<VariantWrapper> writer = variantsAnnotGenerateInputBatch.initWriter(new FileSystemResource(outputFile));
+        writer.open(executionContext);
+        writer.write(Collections.singletonList(variant));
+        assertEquals("20\t60344\t60348\tG/A\t+", readLine());
+        writer.close();
+    }
+
+
+    private void insertDocuments() {
+        collection().insert(constructDbo(variantWithAnnotation));
+        collection().insert(constructDbo(variantWithoutAnnotation));
+    }
+
+    private DBCollection collection() {
+        return mongoClient.getDB(databaseName).getCollection(collectionName);
+    }
+
+    private DBObject constructDbo(String variant) {
+        return (DBObject) JSON.parse(variant);
+    }
+
+    /*
+    * Read a line from the output file, if the reader has not been created,
+    * recreate. This method is only necessary because running the tests in a
+    * UNIX environment locks the file if it's open for writing.
+    *
+    * The variant list should be compressed.
+    */
+    private String readLine() throws IOException {
+        if (reader == null) {
+            reader = new BufferedReader(new FileReader(outputFile));
+        }
+
+        return reader.readLine();
+    }
+
+
+    private final String variantWithoutAnnotation = "{\n" +
```

Comment:
Extracting the 2 variables below to plain file into test/resources folder would simplify making any changes to test input.